### PR TITLE
Added metrics to self-healing logic

### PR DESCRIPTION
**Pull request by begmaroman**
Metrics endpoint: `...:2112/metrics`

### CODE REVIEW

**Review by JekaMas — DISMISSED** [review]

**Review by 0xKrishna — APPROVED** [review]

**Review by JekaMas — APPROVED** [review]

**Comment by temaniarpit27** [comment]
@begmaroman have you checked this on one of the machines that the data if going to Datadog and you are able to view it and apply all kinds of filters?

**Comment by begmaroman** [comment]
@temaniarpit27 I've tested it locally only. From my understanding, we will need to pull metrics from `/metrics` endpoint and then be able to create dashboards. Am I right here?

**Comment by temaniarpit27** [comment]
I think these metrics should also be captured in datadog
we might need to change the datadog agent file but not sure
@vcastellm can you please comment?

**Comment by begmaroman** [comment]
@temaniarpit27 I see we are doing the same on the bor side but using opentel. The HTTP endpoint that exposes metrics data is the same. Seems like we need to adjust datadog agent. Not sure if it could be done within this PR.

**Comment by begmaroman** [comment]
@temaniarpit27 as discussed in Slack, we need to add the daemon metrics config to this list: https://github.com/maticnetwork/ansible/blob/master/playbooks/datadog_agent_blockchain_nodes.yml#L59. Currently we are getting metrics from the heimdall daemon and bor, but not bridge since it does not have any metrics apart from ones defined within this PR.

**Comment by temaniarpit27** [comment]
> @temaniarpit27 as discussed in Slack, we need to add the daemon metrics config to this list: https://github.com/maticnetwork/ansible/blob/master/playbooks/datadog_agent_blockchain_nodes.yml#L59. Currently we are getting metrics from the heimdall daemon and bor, but not bridge since it does not have any metrics apart from ones defined within this PR.

Can we create a linear task for this and inform devops team so that they can pick this up?

**Review by temaniarpit27 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,16 +5,20 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+
       - name: Install Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v3
         with:
-          go-version: 1.13
+          go-version: 1.17.x
```
Comment (vcastellm) [inline .github/workflows/ci.yml:13]: can we use `1.18.x` as in bor?

**Review by vcastellm — APPROVED** [review]
LGTM aside from the Go version

**Comment by begmaroman** [comment]
@temaniarpit27 done, shared a talk link in Slack

File: .github/workflows/ci.yml (hunk 1 above)
Comment (begmaroman) [reply .github/workflows/ci.yml]: Done

**Review by temaniarpit27 — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]

**Pull request merged**